hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/ClientInvocationTest.java

```diff
@@ -115,6 +116,12 @@ public void executionCallback_TooLongThrowableStackTrace() {
         }
     }
 
+    // This test is now failing. Reason is that failure to schedule an execution callback
+    // resulted in the callback's `onFailure` method being called, with the `RejectedExecutionException`
+    // as argument. This seems wrong but maybe some behaviour/functionality (especially related to shutdown)
```

Comment:
The kind of invariants we all love, probably for sure invariants in all of some cases :)